crates/vim/src/vim.rs

```diff
@@ -635,30 +635,23 @@ impl Vim {
         cx.emit(VimEvent::Focused);
         self.sync_vim_settings(cx);
 
-        if let Some(old_vim) = Vim::globals(cx).vim() {
-            if old_vim.entity_id() == cx.view().entity_id() {
-                return;
-            } else {
-                if VimSettings::get_global(cx).toggle_relative_line_numbers {
+        if VimSettings::get_global(cx).toggle_relative_line_numbers {
+            if let Some(old_vim) = Vim::globals(cx).focused_vim() {
+                if old_vim.entity_id() != cx.view().entity_id() {
                     old_vim.update(cx, |vim, cx| {
                         vim.update_editor(cx, |_, editor, cx| {
                             editor.set_relative_line_number(None, cx)
                         });
                     });
-                    self.update_editor(cx, |_, editor, cx| {
-                        editor.set_relative_line_number(Some(true), cx)
-                    });
                 }
-                Vim::globals(cx).vim = Some(cx.view().downgrade());
-            }
-        } else {
-            if VimSettings::get_global(cx).toggle_relative_line_numbers {
-                self.update_editor(cx, |_, editor, cx| {
-                    editor.set_relative_line_number(Some(true), cx)
-                });
             }
-            Vim::globals(cx).vim = Some(cx.view().downgrade());
+
+            self.update_editor(cx, |vim, editor, cx| {
+                let is_relative = vim.mode != Mode::Insert;
+                editor.set_relative_line_number(Some(is_relative), cx)
```

Comment:
If written here, it will cause the command to return and forcibly reset, resulting in the toggle command failure.